metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -849,6 +881,47 @@ def _process_view(
             dataset_name=dataset_name,
         )
 
+    def _process_snapshot(
+        self,
+        snapshot: BigqueryTableSnapshot,
+        columns: List[BigqueryColumn],
+        project_id: str,
+        dataset_name: str,
+    ) -> Iterable[MetadataWorkUnit]:
+        table_identifier = BigqueryTableIdentifier(
+            project_id, dataset_name, snapshot.name
+        )
+
+        self.report.snapshots_scanned += 1
+
+        if not self.config.table_snapshot_pattern.allowed(
+            table_identifier.raw_table_name()
+        ):
+            self.report.report_dropped(table_identifier.raw_table_name())
+            return
+
+        if self.config.include_table_lineage or self.config.include_usage_statistics:
```

Comment:
it feels like we have similar conditionals scattered all over the place. I'd like to refactor some stuff here e.g.

```suggestion
        if self.store_table_refs:
```

```
@property
def store_table_refs(self):
  return self.config.include_table_lineage or self.config.include_usage_statistics
```

that way, we can use `store_table_refs` everywhere. It doesn't need to be called that by the way - as long as the name is clear. This applies a couple other combinations of config flags / complex if statements in this source as well